contentcuration/contentcuration/static/js/edit_channel/sync/views.js

```diff
@@ -0,0 +1,536 @@
+var Backbone = require("backbone");
+var _ = require("underscore");
+var BaseViews = require("edit_channel/views");
+var Models = require("edit_channel/models");
+require("sync.less");
+
+/*********** MODAL CONTAINER FOR SYNC OPERATION ***********/
+var TempSyncModalView = BaseViews.BaseModalView.extend({
+    template: require("./hbtemplates/temp_sync_modal.handlebars"),
+    modal: true,
+
+    initialize: function(options) {
+        this.onsync = options.onsync;
+        this.collection = options.collection;
+        this.selected_options = {};
+        this.render(this.close, {});
+    },
+    events: {
+        'click .category_check' : 'handle_selection',
+        "click #sync_content_button" : "sync_content"
+    },
+    sync_content:function(){
+        var self = this;
+        this.display_load("Syncing Content...", function(resolve, reject){
+           window.current_channel.sync_channel(self.selected_options).then(function(synced){
+                self.onsync(synced);
+                self.close();
+                resolve(true)
+            }).catch(reject);
+        });
+    },
+    get_selected_options: function(){
+        this.selected_options.details = this.$("#check_details").is(':checked');
+        this.selected_options.files = this.$("#check_files").is(':checked');
+        this.selected_options.tags = this.$("#check_tags").is(':checked');
+        this.selected_options.assessment_items = this.$("#check_assessment_items").is(':checked');
+    },
+    handle_selection: function(){
+        this.get_selected_options();
+        if(_.chain(this.selected_options).values().some(function(v){return v;}).value()){
+            this.$("#sync_content_button").prop("disabled", false);
+            this.$("#sync_content_button").removeClass("disabled");
+        } else {
+            this.$("#sync_content_button").prop("disabled", true);
+            this.$("#sync_content_button").addClass("disabled");
+        }
+    }
+});
+
+
+/*********** MODAL CONTAINER FOR SYNC OPERATION ***********/
+var SyncModalView = BaseViews.BaseModalView.extend({
+    template: require("./hbtemplates/sync_modal.handlebars"),
+    modal: true,
+
+    initialize: function(options) {
+        this.render(this.close, {});
+        this.move_view = new SyncView({
+            el: this.$(".modal-body"),
+            onsync: options.onsync,
+            collection : options.collection,
+            modal : this,
+            model:this.model
+        });
+    }
+});
+
+/*********** VIEW FOR SYNC OPERATION ***********/
```

Comment:
modal is more of the wrapper for the sync nodes operation (creates the pop-up) whereas SyncView is what the user actually interacts with